ExotischNederland/Menus/GameMenu.cs

```diff
@@ -0,0 +1,211 @@
+using System;
+using System.Collections.Generic;
+using ExotischNederland.Models;
+
+namespace ExotischNederland.Menus
+{
+    internal class GameMenu
+    {
+        private readonly User authenticatedUser;
+        private Dictionary<string, string> menuItems = new Dictionary<string, string>();
+
+        public GameMenu(User _authenticatedUser)
+        {
+            this.authenticatedUser = _authenticatedUser;
+        }
+
+        public Dictionary<string, string> GetMenuItems()
+        {
+            Dictionary<string, string> menuItems = new Dictionary<string, string>();
+
+            // Add game management options if the user has the "Beheerder" role
+            if (authenticatedUser.Permission.CanManageGames())
+            {
+                menuItems.Add("createGame", "Create a new game");
+                menuItems.Add("editGame", "Edit an existing game");
+                menuItems.Add("deleteGame", "Delete a game");
+            }
+
+            // Add game-playing option for "Familie" and "Kinderen" roles
+            if (authenticatedUser.Permission.CanPlayGames())
+            {
+                menuItems.Add("playGame", "Play a game");
+            }
+
+            menuItems.Add("back", "Return to main menu");
+            return menuItems;
+        }
+
+        public void Show()
+        {
+            menuItems = this.GetMenuItems();
+            while (true)
+            {
+                string selected = Helpers.MenuSelect(this.menuItems, true);
+
+                if (selected == "createGame")
+                {
+                    CreateGame();
+                }
+                else if (selected == "editGame")
+                {
+                    EditGame();
+                }
+                else if (selected == "deleteGame")
+                {
+                    DeleteGame();
+                }
+                else if (selected == "playGame")
+                {
+                    PlayGame();
+                }
+                else if (selected == "back")
+                {
+                    break;
```

Comment:
return does the same, but is clearer. If at any point more code would be added below the while loop the back functionality wouldn't break